### DISCUSSION TITLE
How to create a reusable component with more than one field?

### DISCUSSION THREAD

**Discussion by rudzky** [react-hook-form/react-hook-form#11908] [Q&A]
Hello,
I've created a component `<AddressFields />` which contains 3 inputs: street, zip_code and city. I'd like to use this component in few different forms that have this address fields in the schema. 
I tried to create props like that:   
```ts
type Address = {
  street: string;
  city: string;
  zip_code: string;
};

interface Props<T extends Address> {
  control: Control<T>;
}
```
but then inside the actual <AddressFields /> component I can't use controller:
```ts
const streetController = useController({
  name: "street",  // Type „street” cannot be assigned to „Path<T>”.
  defaultValue: "",
  control,
});
```
How to make this component reusable in every form that contains this address fields? 😄

**Comment by LTsCreed** [comment]
Maybe this is the solution you are looking for - https://github.com/orgs/react-hook-form/discussions/9345
